### PR TITLE
SQL: Add support for SYS GEOMETRY_COLUMNS

### PR DESCRIPTION
**Pull request by imotov**
Adds support for SYS GEOMETRY_COLUMNS, which returns the same
information as `SELECT * FROM GEOMETRY_COLUMNS` command in the standard
implementation.

Relates #29872

### CODE REVIEW

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysGeometryColumns.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            String name = entry.getKey();
+            name = prefix != null ? prefix + "." + name : name;
+            EsField field = entry.getValue();
+            DataType type = field.getDataType();
+            if ((columnMatcher == null || columnMatcher.matcher(name).matches()) && type == GEO_SHAPE) {
+                rows.add(asList(clusterName,  // catalog
+                    null,                 // schema is not supported
+                    indexName,            // table name
+                    name,                 // column name
+                    1,                    // storage type
```
Comment (costin) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysGeometryColumns.java]: Can you please put these constants into something like `GeoUtils` so `DataTypes`? I went with a similar approach in #30418 for readability and maintenance.

**Review by costin — APPROVED** [review]
It's worth adding a parsing test as well (similar to `SysTableTypesTests` or any other under the `command.sys` package) just to make sure the grammar works correctly.

Otherwise LGTM.

By the way, how is the consumer of this command? What are the JDBC/ODBC API calls that this maps to?

**Comment by imotov** [comment]
> By the way, how is the consumer of this command? What are the JDBC/ODBC API calls that this maps to?

@costin that is basically equivalent to `SELECT * FROM GEOMETRY_COLUMNS`, but since we cannot really select from in-memory structures, I had to make it a SYS command. Can you think of a better way of handling this?

**Comment by imotov** [comment]
Discussed this with @costin and @bpintea. It seems that no drivers are using `GEOMETRY_COLUMNS` at the moment we might be able to skip implementing it all together. I am going to close this PR for now. We might need to reopen and merge it later on if needed.

**Pull request closed without merging**